hack/ci/windows.ps1

```diff
@@ -409,7 +409,8 @@ Try {
     # Redirect to a temporary location. 
     $TEMPORIG=$env:TEMP
     $env:TEMP="$env:TESTRUN_DRIVE`:\$env:TESTRUN_SUBDIR\CI-$COMMITHASH"
-    $env:LOCALAPPDATA="$TEMP\localappdata"
+    $env:TMP=$env:TEMP
```

Comment:
According to https://golang.org/pkg/os/#TempDir, `On Windows, it uses GetTempPath, returning the first non-empty value from %TMP%, %TEMP%, %USERPROFILE%, or the Windows directory. ` So in the context of Go, shouldn't the value for %TEMP% be enough?

Reply:
Hmm. To be honest I'm not 100% if that `TMP` variable actually takes effect or did I have `TEMP` redirection missing when I was debugging this as it took for while me to notice that `LOCALAPPDATA` is actually pointing to `\localappdata` as variable name is incorrect and that causes those vet errors like
```12:41:13 vet.exe: in html, can't import facts for package "sync": open \localappdata\go-build\e3\e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855-d: The system cannot find the path specified.``` which example #39193 prints out.

But as those CI machines are super tricky to debug without having access to them I would prefer to redirect both of those just in case or do you want that I change this just modifying LOCALAPPDATA location?

Reply:
Just modifying LOCALAPPDATA to set the correct value sounds good.

Reply:
@ddebroy OK. Modified like that. Thanks for you feedback :smile: